util/reggen/reg_top.sv.tpl

```diff
@@ -731,8 +731,8 @@ ${finst_gen(sr, field, finst_name, fsig_name, fidx)}
   assign addrmiss = (reg_re || reg_we) ? ~|addr_hit : 1'b0 ;
 % if racl_support:
   // A valid address hit but failed the RACL check
-  assign racl_error_o = tl_i.a_valid &
-                        (|addr_hit) & ~(|(addr_hit & (racl_addr_hit_read | racl_addr_hit_write)));
```

Comment:
This will error in the event that it is not a permitted read and not a permitted write, i.e. an error will also be raised when there is no access at all.
I think you're aiming for:
```
assign racl_error_o = |addr_hit & ((reg_re & ~|racl_addr_hit_read) | (reg_we & ~|racl_addr_hit_write));
```

Reply:
Right. Thanks for the help. I just rebased and updated.